railties/lib/rails/test_unit/test_requirer.rb

```diff
@@ -15,7 +15,7 @@ def require_files(patterns)
       private
         def expand_patterns(patterns)
           patterns.map do |arg|
-            arg = arg.gsub(/:(\d+)?$/, '')
+            arg = arg.gsub(/(:\d*)+?$/, '')
```

Comment:
I think we should go with correct syntax `foo.rb:67`, the wrong syntax will throw error so user can correct :)